File: client/src/data/entities/raffle-event.ts
import { CreateEventModel } from './event.ts';

export interface RaffleEventExtras {
  entryCost: number,
  prizeDrawDate: string,
  totalTickets?: number,
  totalDonations?: number
}

export interface CreateRaffleEvent extends CreateEventModel {
  entryCost: number,
  prizeDrawDate: string
}

File: API/Entities/RaffleEvent.cs
namespace API.Entities;

public class RaffleEvent : DeletableEntity
{
    public required string Id { get; set; }
    
    public required string EventId { get; set; }    
    public Event? Event { get; set; }
    
    public DateTime? PrizeDrawDate { get; set; }
    public int EntryCost { get; set; }

    public int TotalTickets { get; set; }
    public int TotalDonations { get; set; }

    public List<RaffleEntry> Entries { get; set; } = [];
    public List<RafflePrize> Prizes { get; set; } = [];
}